cmd/snaptel/task.go

```diff
@@ -115,6 +115,20 @@ func stringValToInt(val string) (int, error) {
 	return parsedField, nil
 }
 
+// stringValToInt parses the input (string) as an unsigned integer value (and returns that uint value
+// to the caller or an error if the input value cannot be parsed as an unsigned integer)
+func stringValToUint(val string) (uint, error) {
+	valInt, err := stringValToInt(val)
+	if err != nil {
+		return uint(0), err
+	}
+	if valInt < 0 {
+		return uint(0), fmt.Errorf("Value '%v' is a negative number and cannot be parsed as an unsigned integer", val)
+	}
+	// return the unsigned integer equivalent of the input string and a nil error (indicating success)
+	return uint(valInt), err
+}
```

Comment:
Not sure but seems to me that max here is MAX_UINT/2, right? Meaning that when we advertise count as uint it may not take values bigger than MAX_UINT/2